spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -82,7 +91,58 @@ public Object resolve(MethodParameter methodParameter, List<String> words) {
 
 	@Override
 	public ParameterDescription describe(MethodParameter parameter) {
-		throw new UnsupportedOperationException();
+		Parameter jlrParameter = parameter.getMethod().getParameters()[parameter.getParameterIndex()];
+		CliOption option = jlrParameter.getAnnotation(CliOption.class);
+		ParameterDescription result = ParameterDescription.outOf(parameter);
+		result.help(option.help());
+		List<String> keys = Arrays.asList(option.key());
+		result.keys(keys.stream()
+				.filter(key -> !key.isEmpty())
+				.map(key -> CLI_PREFIX + key)
+				.collect(Collectors.toList()));
+		Optional<String> defaultValue = defaultValueFor(option, result.keys());
+		if (defaultValue.isPresent()) {
+			result.defaultValue(defaultValue.get());
+		}
+		boolean containsEmptyKey = keys.contains("");
+		result.mandatoryKey(!containsEmptyKey);
```

Comment:
Hmm, this is one I've been debating a bit in my head and need input.

Decided to revert it to what I had before where the key is mandatory if the empty key isn't provided as part of the possible set of keys.

However, the reason I'm still unsure about this is due to the unspecifiedDefaultValue and the possible combinations with mandatory.

Can a mandatory CliOption have an unspecifiedDefaultValue? (or better put, should it be allowed?) If the answer is no, I believe the current logic should handle what we want.